foundry/https_upload.py

```diff
@@ -0,0 +1,248 @@
+            adding/removing folders, etc).
+        endpoint_auth_clients (Dict[str, AuthClient]): Dict of `endpoint_id` : `AuthClient` keypairs. AuthClients used
+            for Globus Auth functionality within endpoint-specific scopes using Globus Connect Server (ie accessing
+            or altering data on a specific endpoint).
+        local_data_path (str): The path to the local data to upload. Can be relative or absolute.
+        dest_path (str): The path to the destination folder on the Globus endpoint. Default is "/tmp".
+        endpoint_id (str): A UUID designating the exact Globus endpoint. Can be obtained via the Globus Web UI or
+            the SDK. Default is the NCSA UUID "82f1b5c6-6e9b-11e5-ba47-22000b92c6ec".
+    Returns
+    -------
+        (str): Globus data source URL (ie the URL that points to the data on a Globus endpoint)
+    """
+
+    # get URL for Globus endpoint location
+    endpoint = transfer_client.get_endpoint(endpoint_id)  # gets info for NCSA endpoint
+    https_base_url = endpoint["https_server"]
+
+    # Submit data (folders of files or an independent file) to be written to endpoint
+    if os.path.isdir(local_data_path):
+        _upload_folder(transfer_client, endpoint_auth_clients, local_data_path, https_base_url, dest_path, endpoint_id)
+    elif os.path.isfile(local_data_path):
+        _upload_file(endpoint_auth_clients[endpoint_id], local_data_path, https_base_url, dest_path)
+    else:
+        raise IOError(f"Data path '{local_data_path}' is of unknown type")
+
+    # return the data source URL for publication to MDF
+    return _make_globus_link(endpoint_id, dest_path)
+
+
+def _upload_folder(transfer_client: TransferClient, endpoint_auth_clients: Dict[str, AuthClient], local_data_path: str,
+                   https_base_url: str, parent_dest_path: str, endpoint_id: str) -> List[Dict[str, Any]]:
+    """Upload a folder to a Globus endpoint using HTTPS
+    Args:
+        transfer_client (TransferClient): Globus client authorized for Globus Transfers (ie moving data on endpoint,
+            adding/removing folders, etc).
+        endpoint_auth_clients (Dict[str, AuthClient]): Dict of `endpoint_id` : `AuthClient` keypairs. AuthClients used
+            for Globus Auth functionality within endpoint-specific scopes using Globus Connect Server (ie accessing
+            or altering data on a specific endpoint).
+        local_data_path (str): The path to the local data to upload. Can be relative or absolute.
+        https_base_url (str): The URL for a given Globus endpoint.
+        parent_dest_path (str): The path to the parent folder to be written to on the given endpoint. The contents
+            of "local_data_path" will be written here, including subdirectories.
+        endpoint_id (str): The UUID designating the exact Globus endpoint. Can be obtained via the Globus Web UI or
+            the SDK. This must be the same endpoint pointed to by the https_base_url.
+    Returns
```

Comment:
All of the doc strings mix NumPy and Google Styles. ("Args:" is Google, "Return\n-----" is NumPy). I see that the whole library does this, so fix it here if you want.